projects/ad355xr_iio/ci_build.groovy

```diff
@@ -0,0 +1,65 @@
+		buildMap = [
+			PLATFORM_NAME: ['SDP_K1', 'NUCLEO_L552ZE_Q', 'DISCO_F769NI'],
+			ACTIVE_DEVICE: ['DEV_AD3541R','DEV_AD3542R_12','DEV_AD3542R_16','DEV_AD3551R','DEV_AD3552R'],
+			COM_TYPE: ['USE_PHY_COM_PORT','USE_VIRTUAL_COM_PORT'],
+			SDRAM: ['USE_SDRAM', 'NO_SDRAM']
+		]
+	} else {
+		// This map is for building targets that is always build
+		buildMap = [
+			PLATFORM_NAME: ['SDP_K1'],
+			ACTIVE_DEVICE: ['DEV_AD3552R'],
+			COM_TYPE: ['USE_VIRTUAL_COM_PORT'],
+			SDRAM: ['NO_SDRAM']
+		]
+	}
+
+	// Get the matrix combination and filter them to exclude unrequired matrixes
+	List buildMatrix = buildInfo.getMatrixCombination(buildMap).findAll { axis ->
+		// Skip 'all platforms except SDP-K1 + SDRAM and + VCOM and + devices other than ad3552r'
+		!(axis['PLATFORM_NAME'] != 'SDP_K1' && axis['SDRAM'] == 'USE_SDRAM') &&
+		!(axis['PLATFORM_NAME'] != 'SDP_K1' && axis['COM_TYPE'] == 'USE_VIRTUAL_COM_PORT') &&
+		!(axis['PLATFORM_NAME'] != 'SDP_K1' && axis['ACTIVE_DEVICE'] != 'DEV_AD3552R')
+	}
+	
+	return buildMatrix
+}
+
+def doBuild(Map config =[:], projectName) {
+	stage("Build-${config.PLATFORM_NAME}-${config.ACTIVE_DEVICE}-${config.COM_TYPE}-${config.SDRAM}") {
+		FIRMWARE_VERSION = buildInfo.getFirmwareVersion()
+
+		echo "^^**^^ ^^**^^ ^^**^^ ^^**^^ ^^**^^ ^^**^^ ^^**^^ ^^**^^ ^^**^^ ^^**^^"     
+		echo "Running on node: '${env.NODE_NAME}'"
+	
+		echo "TOOLCHAIN: ${TOOLCHAIN}"
+		echo "TOOLCHAIN_PATH: ${TOOLCHAIN_PATH}"
+	
+		echo "Building for ${config.PLATFORM_NAME} Platform and ${config.ACTIVE_DEVICE} Device with ${config.COM_TYPE} UART type and ${config.SDRAM}"
+		echo "^^**^^ ^^**^^ ^^**^^ ^^**^^ ^^**^^ ^^**^^ ^^**^^ ^^**^^ ^^**^^ ^^**^^"
+	
+		echo "Starting mbed build..."
+		//NOTE: if adding in --profile, need to change the path where the .bin is found by mbedflsh in Test stage
+		sh "cd projects/${projectName} ; make all LINK_SRCS=n TARGET_BOARD=${config.PLATFORM_NAME} BINARY_FILE_NAME=${config.PLATFORM_NAME}-${config.ACTIVE_DEVICE}-${config.COM_TYPE}-${config.SDRAM} NEW_CFLAGS+=-DPLATFORM_NAME=${config.PLATFORM_NAME} NEW_CFLAGS+=-DFIRMWARE_VERSION=${FIRMWARE_VERSION} NEW_CFLAGS+=-D${config.ACTIVE_DEVICE} NEW_CFLAGS+=-D${config.COM_TYPE} NEW_CFLAGS+=-D${config.SDRAM}"
+		artifactory.uploadFirmwareArtifacts("projects/${projectName}/build","${projectName}")
+		archiveArtifacts allowEmptyArchive: true, artifacts: "projects/${projectName}/build/*.bin, projects/${projectName}/build/*.elf"
+		stash includes: "projects/${projectName}/build/*.bin, projects/${projectName}/build/*.elf", name: "${config.PLATFORM_NAME}-${config.ACTIVE_DEVICE}-${config.COM_TYPE}-${config.SDRAM}"
+		sh "cd projects/${projectName} ; make reset LINK_SRCS=n TARGET_BOARD=${config.PLATFORM_NAME}"
+	}
+}
+
+def getTestMatrix(projectName) {
+	// TODO: Add the test sequence once CI test setup is up
+	List testMatrix = []
+	return testMatrix
+}
+
+def doTest(Map config =[:], projectName) {
+	// TODO: Add the test sequence once CI test setup is up
```

Comment:
There shouldn't  be any TODOs in the code we are releasing. It is a part of the firmware release checklist

Reply:
But still there is no pyadi-iio drivers and example are not published on pyadi-iio repo. Now we are only releasing firmware source code.

Reply:
Should we not be adding the pyadi drivers here until they are up on pyadi-iio repo?

Reply:
yes.